### PR TITLE
Code updates

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

For your consideration, this PR includes several simplifications/improvements to code as suggested by PHPStorm inspection tools.

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/AbstractAdapter.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ static function ($deferred, $namespace, &$expiredIds, $getId, $defaultLifetime)
                 foreach ($deferred as $key => $item) {
                     $key = (string) $key;
                     if (null === $item->expiry) {
-                        $ttl = 0 < $defaultLifetime ? $defaultLifetime : 0;
+                        $ttl = max(0, $defaultLifetime);
```
Comment (derrabus) [inline src/Symfony/Component/Cache/Adapter/AbstractAdapter.php]: This is mostly a readability improvement and I agree with switching to `max()` in most of the places where you've applied this change.

However, the ternary expression might be a tiny bit faster than a function call. Given that this particular piece code might be executed on a hot path, I'd revert this kind of change on the cache adapters.

File: src/Symfony/Component/Mailer/Bridge/Infobip/Tests/Transport/InfobipApiTransportTest.php [hunk 2]
```diff
@@ -39,10 +39,7 @@ public function request(string $method, string $url, array $options = []): Respo
                 // The only purpose of this method override is to record the request body as a string
                 // It's impossible to get the generated body when using a generator as a body
                 if (isset($options['body']) && $options['body'] instanceof \Generator) {
-                    $body = '';
-                    foreach ($options['body'] as $data) {
-                        $body .= $data;
-                    }
+                    $body = implode('', $options['body']);
```
Comment (derrabus) [inline src/Symfony/Component/Mailer/Bridge/Infobip/Tests/Transport/InfobipApiTransportTest.php]: This won't work. `implode()` is made for arrays, `$options['body']` is a generator (see the line above your change).

File: src/Symfony/Component/Mailer/Tests/Transport/Smtp/Stream/AbstractStreamTest.php [hunk 3]
```diff
@@ -21,10 +21,7 @@ class AbstractStreamTest extends TestCase
      */
     public function testReplace(string $expected, string $from, string $to, array $chunks)
     {
-        $result = '';
-        foreach (AbstractStream::replace($from, $to, $chunks) as $chunk) {
-            $result .= $chunk;
-        }
+        $result = implode('', AbstractStream::replace($from, $to, $chunks));
```
Comment (derrabus) [inline src/Symfony/Component/Mailer/Tests/Transport/Smtp/Stream/AbstractStreamTest.php]: Same here: we probably won't get an array here, so `implode()` won't work.

File: src/Symfony/Component/Mime/Part/SMimePart.php [hunk 4]
```diff
@@ -52,10 +52,7 @@ public function bodyToString(): string
             return $this->body;
         }
 
-        $body = '';
-        foreach ($this->body as $chunk) {
-            $body .= $chunk;
-        }
+        $body = implode('', $this->body);
```
Comment (derrabus) [inline src/Symfony/Component/Mime/Part/SMimePart.php]: Same, `$this->body` can be any iterable while `implode()` only works on arrays.

File: src/Symfony/Component/Mime/RawMessage.php [hunk 5]
```diff
@@ -32,10 +32,7 @@ public function toString(): string
             return $this->message;
         }
 
-        $message = '';
-        foreach ($this->message as $chunk) {
-            $message .= $chunk;
-        }
+        $message = implode('', $this->message);
```
Comment (derrabus) [inline src/Symfony/Component/Mime/RawMessage.php]: Same.

File: src/Symfony/Component/Mime/Tests/Crypto/SMimeTestCase.php [hunk 6]
```diff
@@ -41,15 +41,9 @@ protected function normalizeFilePath(string $path): string
...
-        $string = '';
-
         // Can't use iterator_to_array as the generators are merged internally,
         // leading to overwritten keys
-        foreach ($iterable as $chunk) {
-            $string .= $chunk;
-        }
-
-        return $string;
+        return implode('', $iterable);
```
Comment (derrabus) [inline src/Symfony/Component/Mime/Tests/Crypto/SMimeTestCase.php]: Same.

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 7]
```diff
@@ -167,10 +167,7 @@ public function provideNormalizeCallbacks()
...
                 [
                     'bar' => function (array $bars) {
-                        $result = '';
-                        foreach ($bars as $bar) {
-                            $result .= $bar->bar;
-                        }
+                        $result = implode('', $bars);
 
                         return $result;
                     },
```
Comment (derrabus) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: After your simplification, `$result` can be inlined and then the function can be further simplified to an arrow function.

```php
static fn (array $bars) => implode('', $bars)
```

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 8]
```diff
@@ -229,10 +226,7 @@ public function provideDenormalizeCallbacks(): array
...
```
Comment (derrabus) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: Same here.

File: src/Symfony/Component/Validator/Constraints/UniqueValidator.php [hunk 9]
```diff
@@ -60,11 +60,7 @@ public function validate(mixed $value, Constraint $constraint): void
 
     private function getNormalizer(Unique $unique): callable
     {
-        if (null === $unique->normalizer) {
-            return static fn ($value) => $value;
-        }
-
-        return $unique->normalizer;
+        return $unique->normalizer ?? static fn($value) => $value;
```
Comment (derrabus) [inline src/Symfony/Component/Validator/Constraints/UniqueValidator.php]: CS

```suggestion
        return $unique->normalizer ?? static fn ($value) => $value;
```

File: src/Symfony/Component/Cache/Adapter/AbstractAdapter.php (hunk 1 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Cache/Adapter/AbstractAdapter.php]: Thanks. This is now reverted.

File: src/Symfony/Component/Mailer/Bridge/Infobip/Tests/Transport/InfobipApiTransportTest.php (hunk 2 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Mailer/Bridge/Infobip/Tests/Transport/InfobipApiTransportTest.php]: Sorry! The automatic tool totally fooled me on this. I reverted this and all the other occurrences. Thanks.

**Review by javiereguiluz — COMMENTED** [review]
Thanks Alexander for the review. I made all the changes and fixes that you suggested.

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php (hunk 8 above)
Comment (derrabus) [reply src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: Sorry, with "Same here." I was referring to https://github.com/symfony/symfony/pull/53119#discussion_r1429754661: The change is correct, but we can simplify the code further to a single-line arrow function.

Comment (derrabus) [reply src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: My bad. 🙈

File: src/Symfony/Component/Yaml/Escaper.php [hunk 10]
```diff
@@ -80,7 +80,7 @@ public static function requiresSingleQuoting(string $value): bool
 
         // Determines if the PHP value contains any single characters that would
         // cause it to require single quoting in YAML.
-        return 0 < preg_match('/[ \s \' " \: \{ \} \[ \] , & \* \# \?] | \A[ \- ? | < > = ! % @ ` \p{Zs}]/xu', $value);
+        return 0 < preg_match('/[ \s\'"\:\{\}\[\],&\*\#\?] | \A[ \-?|<>=!%@`\p{Zs}]/xu', $value);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Yaml/Escaper.php]:
```suggestion
        return 0 < preg_match('/[\s\'"\:\{\}\[\],&\*\#\?] | \A[\-?|<>=!%@`\p{Zs}]/xu', $value);
```

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 11]
```diff
@@ -166,14 +166,7 @@ public function provideNormalizeCallbacks()
             ],
             'Collect a property' => [
                 [
-                    'bar' => function (array $bars) {
-                        $result = '';
-                        foreach ($bars as $bar) {
-                            $result .= $bar->bar;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: are we sure about this change? we don't read `->bar` directly in the new version

Comment (derrabus) [reply src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: Oh, you're right! I totally missed that.

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php (hunk 8 above)
Comment (derrabus) [reply src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: NVM, Nicolas is right. Reverting this change is correct.

File: src/Symfony/Component/Form/Util/OrderedHashMapIterator.php [hunk 12]
```diff
@@ -98,11 +98,7 @@ public function next(): void
 
     public function key(): mixed
     {
-        if (null === $this->key) {
-            return null;
-        }
-
-        return $this->key;
+        return $this->key ?? null;
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]:
```suggestion
        return $this->key;
```

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php (hunk 11 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: Reverted this. Sorry.

File: src/Symfony/Component/Form/Util/OrderedHashMapIterator.php (hunk 12 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]: Done. Thanks.

File: src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyOptions.php [hunk 13]
```diff
@@ -67,16 +67,12 @@ public function setTitle(string $title): self
...
-            case Notification::IMPORTANCE_URGENT:
-                return $this->setPriority(self::PRIORITY_URGENT);
-            case Notification::IMPORTANCE_HIGH:
-                return $this->setPriority(self::PRIORITY_HIGH);
-            case Notification::IMPORTANCE_LOW:
-                return $this->setPriority(self::PRIORITY_LOW);
-            default:
-                return $this->setPriority(self::PRIORITY_DEFAULT);
-        }
+        return match ($priority) {
```
Comment (diimpp) [inline src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyOptions.php:70]: Just in case, `setPriority` allows for one more priority, which is not handled here: `PRIORITY_MIN`

image

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 14]
```diff
@@ -165,6 +165,13 @@ public function provideNormalizeCallbacks()
                 ['bar' => '10-09-2011 06:30:00', 'foo' => null],
             ],
             'Collect a property' => [
+                [
+                    'bar' => static fn (array $bars) => implode('', $bars),
+                ],
+                [new CallbacksObject('baz'), new CallbacksObject('quux')],
+                ['bar' => 'bazquux', 'foo' => null],
+            ],
+            'Count a property' => [
```
Comment (diimpp) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: Something happened over here, `Count a property` should have `bar` with `\count`, but it now has previous `implode` version.

image

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 15]
```diff
@@ -165,6 +165,13 @@ public function provideNormalizeCallbacks()
                 ['bar' => '10-09-2011 06:30:00', 'foo' => null],
             ],
             'Collect a property' => [
+                [
+                    'bar' => static fn (array $bars) => implode('', $bars),
```
Comment (diimpp) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php:169]:
```suggestion
                    'bar' => static fn (array $bars) => \implode('', $bars),
```

File: src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php [hunk 16]
```diff
@@ -89,7 +89,7 @@ public function supportsNormalization(mixed $data, string $format = null, array
      */
     public function denormalize(mixed $data, string $type, string $format = null, array $context = []): \SplFileInfo
     {
-        if (null === $data || !preg_match('/^data:([a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}\/[a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}(;[a-z0-9\-]+\=[a-z0-9\-]+)?)?(;base64)?,[a-z0-9\!\$\&\\\'\,\(\)\*\+\,\;\=\-\.\_\~\:\@\/\?\%\s]*\s*$/i', $data)) {
+        if (null === $data || !preg_match('/^data:([a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}\/[a-z0-9][a-z0-9\!\#\$\&\-\^\_\+\.]{0,126}(;[a-z0-9\-]+\=[a-z0-9\-]+)?)?(;base64)?,[a-z0-9\!\$\&\\\'\,\(\)\*\+\;\=\-\.\_\~\:\@\/\?\%\s]*\s*$/i', $data)) {
```
Comment (diimpp) [inline src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php:92]: Only Jesus can review this. :smiling_face_with_tear: Looks like it should be in separate PR and possibly with test case for the current issue.

File: src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php [hunk 17]
```diff
@@ -90,7 +90,7 @@ public function setCharset(string $charset): string
         $prev = $this->charset;
 
         $charset = strtoupper($charset);
-        $charset = null === $charset || 'UTF-8' === $charset || 'UTF8' === $charset ? 'CP1252' : $charset;
+        $charset = 'UTF-8' === $charset || 'UTF8' === $charset ? 'CP1252' : $charset;
```
Comment (diimpp) [inline src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php:93]: As an idea, there is no check for empty charset. Though I didn't check how and when this method is used.

```suggestion
        $charset = '' === $charset || 'UTF-8' === $charset || 'UTF8' === $charset ? 'CP1252' : $charset;
```

File: src/Symfony/Component/Yaml/Dumper.php [hunk 18]
```diff
@@ -169,7 +169,7 @@ private function getBlockIndentationIndicator(string $value): string
         // http://www.yaml.org/spec/1.2/spec.html#id2793979
         foreach ($lines as $line) {
             if ('' !== trim($line, ' ')) {
-                return (' ' === substr($line, 0, 1)) ? (string) $this->indentation : '';
+                return (str_starts_with($line, ' ')) ? (string) $this->indentation : '';
```
Comment (diimpp) [inline src/Symfony/Component/Yaml/Dumper.php]:
```suggestion
                return str_starts_with($line, ' ') ? (string) $this->indentation : '';
```

File: src/Symfony/Component/Yaml/Escaper.php [hunk 19]
```diff
@@ -80,7 +80,7 @@ public static function requiresSingleQuoting(string $value): bool
 
         // Determines if the PHP value contains any single characters that would
         // cause it to require single quoting in YAML.
-        return 0 < preg_match('/[ \s \' " \: \{ \} \[ \] , & \* \# \?] | \A[ \- ? | < > = ! % @ ` \p{Zs}]/xu', $value);
+        return 0 < preg_match('/[\s\'"\:\{\}\[\],&\*\#\?] | \A[\-?|<>=!%@`\p{Zs}]/xu', $value);
```
Comment (diimpp) [inline src/Symfony/Component/Yaml/Escaper.php:83]: I'm not reviewing this either. :sweat_smile:

File: .github/get-modified-packages.php [hunk 20]
```diff
@@ -19,31 +19,15 @@
...
-    }
-
-    throw new \LogicException();
+    return match (true) {
+        str_contains($packageDir, 'Symfony/Bridge/') => 'bridge',
+        str_contains($packageDir, 'Symfony/Bundle/') => 'bundle',
+        preg_match('@Symfony/Component/[^/]+/Bridge/@', $packageDir) => 'component_bridge',
+        str_contains($packageDir, 'Symfony/Component/') => 'component',
+        str_contains($packageDir, 'Symfony/Contracts/') => 'contract',
+        str_ends_with($packageDir, 'Symfony/Contracts') => 'contracts',
```
Comment (diimpp) [inline .github/get-modified-packages.php:28]: I guess it's part of CS as in some place it used and in some not
```suggestion
        \str_contains($packageDir, 'Symfony/Bridge/') => 'bridge',
        \str_contains($packageDir, 'Symfony/Bundle/') => 'bundle',
        \preg_match('@Symfony/Component/[^/]+/Bridge/@', $packageDir) => 'component_bridge',
        \str_contains($packageDir, 'Symfony/Component/') => 'component',
        \str_contains($packageDir, 'Symfony/Contracts/') => 'contract',
        \str_ends_with($packageDir, 'Symfony/Contracts') => 'contracts',
```

File: src/Symfony/Component/Dotenv/Dotenv.php [hunk 21]
```diff
@@ -463,7 +463,7 @@ private function resolveCommands(string $value, array $loadedVars): string
                 throw $this->createFormatException(sprintf('Issue expanding a command (%s)', $process->getErrorOutput()));
             }
 
-            return preg_replace('/[\r\n]+$/', '', $process->getOutput());
+            return rtrim($process->getOutput(), "\n\r");
```
Comment (diimpp) [inline src/Symfony/Component/Dotenv/Dotenv.php:466]:
```suggestion
            return \rtrim($process->getOutput(), "\n\r");
```

File: src/Symfony/Component/Filesystem/Path.php [hunk 22]
```diff
@@ -346,13 +346,13 @@ public static function changeExtension(string $path, string $extension): string
         $extension = ltrim($extension, '.');
 
         // No extension for paths
-        if ('/' === substr($path, -1)) {
+        if (str_ends_with($path, '/')) {
```
Comment (diimpp) [inline src/Symfony/Component/Filesystem/Path.php:349]:
```suggestion
        if (\str_ends_with($path, '/')) {
```

File: src/Symfony/Component/Filesystem/Path.php [hunk 23]
```diff
@@ -346,13 +346,13 @@ public static function changeExtension(string $path, string $extension): string
...
             return $path;
         }
 
         // No actual extension in path
         if (empty($actualExtension)) {
-            return $path.('.' === substr($path, -1) ? '' : '.').$extension;
+            return $path.(str_ends_with($path, '.') ? '' : '.').$extension;
```
Comment (diimpp) [inline src/Symfony/Component/Filesystem/Path.php:355]:
```suggestion
            return $path.(\str_ends_with($path, '.') ? '' : '.').$extension;
```

File: src/Symfony/Component/HttpClient/CurlHttpClient.php [hunk 24]
```diff
@@ -106,7 +106,7 @@ public function request(string $method, string $url, array $options = []): Respo
             \CURLOPT_PROTOCOLS => \CURLPROTO_HTTP | \CURLPROTO_HTTPS,
             \CURLOPT_REDIR_PROTOCOLS => \CURLPROTO_HTTP | \CURLPROTO_HTTPS,
             \CURLOPT_FOLLOWLOCATION => true,
-            \CURLOPT_MAXREDIRS => 0 < $options['max_redirects'] ? $options['max_redirects'] : 0,
+            \CURLOPT_MAXREDIRS => max(0, $options['max_redirects']),
```
Comment (diimpp) [inline src/Symfony/Component/HttpClient/CurlHttpClient.php:109]:
```suggestion
            \CURLOPT_MAXREDIRS => \max(0, $options['max_redirects']),
```

File: src/Symfony/Component/HttpFoundation/Cookie.php [hunk 25]
```diff
@@ -340,7 +340,7 @@ public function getMaxAge(): int
     {
         $maxAge = $this->expire - time();
 
-        return 0 >= $maxAge ? 0 : $maxAge;
+        return max(0, $maxAge);
```
Comment (diimpp) [inline src/Symfony/Component/HttpFoundation/Cookie.php:343]:
```suggestion
        return \max(0, $maxAge);
```

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php [hunk 26]
```diff
@@ -358,8 +358,8 @@ public function send(string $body, array $headers, int $delay = 0, string $messa
         }
 
         if (self::isFifoQueue($this->configuration['queue_name'])) {
-            $parameters['MessageGroupId'] = null !== $messageGroupId ? $messageGroupId : __METHOD__;
-            $parameters['MessageDeduplicationId'] = null !== $messageDeduplicationId ? $messageDeduplicationId : sha1(json_encode(['body' => $body, 'headers' => $headers]));
+            $parameters['MessageGroupId'] = $messageGroupId ?? __METHOD__;
+            $parameters['MessageDeduplicationId'] = $messageDeduplicationId ?? sha1(json_encode(['body' => $body, 'headers' => $headers]));
```
Comment (diimpp) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:362]:
```suggestion
            $parameters['MessageDeduplicationId'] = $messageDeduplicationId ?? sha1(\json_encode(['body' => $body, 'headers' => $headers]));
```

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 27]
```diff
@@ -138,7 +138,7 @@ private function registerHandlers(ContainerBuilder $container, array $busIds): v
                 }
 
                 if (null === $message) {
-                    throw new RuntimeException(sprintf('Invalid handler service "%s": the list of messages to handle is empty.', $serviceId, $r->getName()));
+                    throw new RuntimeException(sprintf('Invalid handler service "%s": the list of messages to handle is empty.', $serviceId));
```
Comment (diimpp) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php:141]:
```suggestion
                    throw new RuntimeException(\sprintf('Invalid handler service "%s": the list of messages to handle is empty.', $serviceId));
```

File: src/Symfony/Component/Mime/CharacterStream.php [hunk 28]
```diff
@@ -111,7 +111,7 @@ public function read(int $length): ?string
             $this->currentPos += $length;
         } else {
             $end = $this->currentPos + $length;
-            $end = $end > $this->charCount ? $this->charCount : $end;
+            $end = min($end, $this->charCount);
```
Comment (diimpp) [inline src/Symfony/Component/Mime/CharacterStream.php:114]:
```suggestion
            $end = \min($end, $this->charCount);
```

File: src/Symfony/Component/Yaml/Escaper.php (hunk 19 above)
Comment (diimpp) [inline src/Symfony/Component/Yaml/Escaper.php:83]:
```suggestion
        return 0 < \preg_match('/[\s\'"\:\{\}\[\],&\*\#\?] | \A[\-?|<>=!%@`\p{Zs}]/xu', $value);
```

**Review by diimpp — COMMENTED** [review]
Didn't find mention in SF CS whenever global functions should be used as `min()` or `\min()`, please ignore/resolve those comments, if this is not necessary change.

File: src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php [hunk 29]
```diff
@@ -175,13 +182,6 @@ public function provideNormalizeCallbacks()
                         return $result;
                     },
                 ],
-                [new CallbacksObject('baz'), new CallbacksObject('quux')],
-                ['bar' => 'bazquux', 'foo' => null],
-            ],
-            'Count a property' => [
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Tests/Normalizer/Features/CallbacksTestTrait.php]: looks like something went wrong when changing this trait (see the failing tests)

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Comment by xabbuh** [comment]
@diimpp Symfony uses `@compiler_optimized` as the value of the [native_function_invocation](https://cs.symfony.com/doc/rules/function_notation/native_function_invocation.html) which means that only a small set of functions which can be optimized by PHP have to be prefixed with a backslash (see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php#L327-L377 for a list of functions).

File: src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php (hunk 17 above)
Comment (javiereguiluz) [reply src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php:93]: I prefer to only make the strictly minimum changes. In this case, we have `$charset = strtoupper($charset);` and we know hat `strtoupper()` only returns `string`, so we can safely remove the `null === $charset` check. Other than that, I'd prefer to not change anything..

File: src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php (hunk 16 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php:92]: This is easier than it looks; it's just that the `\,` was included twice in the regexp.

File: src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyOptions.php (hunk 13 above)
Comment (javiereguiluz) [reply src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyOptions.php:70]: In that case, the `default` value is applied to it, so it's fine.

File: src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php (hunk 16 above)
Comment (GromNaN) [reply src/Symfony/Component/Serializer/Normalizer/DataUriNormalizer.php:92]: LGTM. There's no issue, only a duplicate `\,` in the char list.

**Review by nicolas-grekas — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @javiereguiluz.
